src/ibtracert.c

```diff
@@ -896,9 +909,31 @@ int main(int argc, char **argv)
 
 	node_name_map = open_node_name_map(node_name_map_file);
 
-	if (get_route(argv[0], argv[1]) != 0)
-		IBEXIT("Failed to get route information");
-
+	if (ports_file == NULL) {
+		/* single get_route call when lids/guids on command line */
+		if (get_route(argv[0], argv[1]) != 0)
+			IBEXIT("Failed to get route information");
+	} else {
+		/* multiple get_route calls when reading lids/guids from a file */
+		ports_fd = fopen(ports_file, "r");
+		if (!ports_fd)
+			IBEXIT("cannot open ports-file %s", ports_file);
+
+		while (fgets(portsbuf, sizeof(portsbuf), ports_fd) != NULL) {
+			line_count++;
+			if (portsbuf[0] != '#') {
```

Comment:
Susan,

A follow up patch would be nice to make this work if the comment is not at the beginning of the line...  But this will be fine for now.